dotCMS/src/main/java/com/dotcms/ai/client/openai/OpenAIClient.java

```diff
@@ -0,0 +1,166 @@
+                        jsonRequest.getMethod(),
+                        System.lineSeparator(),
+                        appConfig.toString(),
+                        System.lineSeparator(),
+                        jsonRequest.payloadToString()));
+
+        if (!appConfig.isEnabled()) {
+            AppConfig.debugLogger(OpenAIClient.class, () -> "App dotAI is not enabled and will not send request.");
+            throw new DotAIAppConfigDisabledException("App dotAI config without API urls or API key");
+        }
+
+        final JSONObject payload = jsonRequest.getPayload();
+        final String modelName = Optional
+                .ofNullable(payload.optString(AiKeys.MODEL))
+                .orElseThrow(() -> new DotAIModelNotFoundException("Model is not present in the request"));
+        // TODO: pr-split -> uncomment this line
+        //final Tuple2<AIModel, Model> modelTuple = appConfig.resolveModelOrThrow(modelName, jsonRequest.getType());
+        final Tuple2<AIModel, Model> modelTuple = Tuple.of(null, null);
+        final AIModel aiModel = modelTuple._1;
+
+        if (!modelTuple._2.isOperational()) {
+            AppConfig.debugLogger(
+                    getClass(),
+                    () -> String.format("Resolved model [%s] is not operational, avoiding its usage", modelName));
+            throw new DotAIModelNotOperationalException(String.format("Model [%s] is not operational", modelName));
+        }
+
+        final long sleep = lastRestCall.computeIfAbsent(aiModel, m -> 0L)
+                + aiModel.minIntervalBetweenCalls()
+                - System.currentTimeMillis();
+        if (sleep > 0) {
+            Logger.info(
+                    this,
+                    "Rate limit:"
+                            + aiModel.getApiPerMinute()
+                            + "/minute, or 1 every "
+                            + aiModel.minIntervalBetweenCalls()
+                            + "ms. Sleeping:"
+                            + sleep);
+            Try.run(() -> Thread.sleep(sleep));
+        }
+
+        lastRestCall.put(aiModel, System.currentTimeMillis());
+
+        try (CloseableHttpClient httpClient = HttpClients.createDefault()) {
+            final StringEntity jsonEntity = new StringEntity(payload.toString(), ContentType.APPLICATION_JSON);
+            final HttpUriRequest httpRequest = AIClient.resolveMethod(jsonRequest.getMethod(), jsonRequest.getUrl());
+            httpRequest.setHeader(HttpHeaders.CONTENT_TYPE, MediaType.APPLICATION_JSON);
+            httpRequest.setHeader(HttpHeaders.AUTHORIZATION, "Bearer " + appConfig.getApiKey());
+
+            if (!payload.getAsMap().isEmpty()) {
+                Try.run(() -> ((HttpEntityEnclosingRequestBase) httpRequest).setEntity(jsonEntity));
+            }
+
+            try (CloseableHttpResponse response = httpClient.execute(httpRequest)) {
+                final BufferedInputStream in = new BufferedInputStream(response.getEntity().getContent());
```

Comment:
I think we should have this code already in some place

Reply:
Yes. On `OpenAIRequest` class which is about to disappear in favor of `OpenAIClient`.